components/TextArea/TextArea.jsx

```diff
@@ -31,22 +37,50 @@ class TextArea extends Component {
   constructor(props) {
     super(props);
 
-    const { id, value } = props;
+    const {
+      id,
+      value,
+      autoResizeConfig: { rowsConfig, minRowsConfig, maxRowsConfig },
+    } = props;
 
     this.state = {
       hasDefaultValue: value !== null && value[0],
       currentValue: value,
+      rows: rowsConfig,
+      minRows: minRowsConfig,
+      maxRows: maxRowsConfig,
     };
 
     this._id = id || ID_GENERATOR.next().value;
+    this._fontSize = 16;
+    this._linHeight = 1.5;
   }
 
   onChangeTextArea = ev => {
     const { onChange } = this.props;
+    const { minRows, maxRows } = this.state;
+    const textareaLineHeightInPx = this._fontSize * this._linHeight;
+    const previousRows = ev.target.rows;
```

Comment:
how about of using destructuting here, to make code cleaner?
here my suggestions:
- i updated the contant names
- renamed the event arg
- and i applied a desctructuring in ev.target.
```
  onChangeTextArea = event => {
    const { target } = event;
    const { onChange } = this.props;
    const { minRows, maxRows } = this.state;
    const textareaLineHeightInPx = this._fontSize * this._lineHeight;
    const previousNumberRows = target.rows;
    target.rows = minRows;

    const currentNumberRows = Math.floor(
      target.scrollHeight / textareaLineHeightInPx,
    );
    
    if (currentNumberRows === previousNumberRows) {
      target.rows = currentNumberRows;
    }

    if (currentNumberRows >= maxRows) {
      target.rows = maxRows;
      target.scrollTop = target.scrollHeight;
    }

    const inputValue = event.currentTarget.value;
    this.setState({
      currentValue: inputValue,
      hasDefaultValue: null,
      rows: currentNumberRows < maxRows ? currentNumberRows : maxRows,
    });

    onChange(event);
  };
```